### PR TITLE
Show current locale attachments only

### PR DESCRIPTION
**Pull request by gpeng**
This will make Publication documents present only the attachments that match the current locale of the document.

This is required to ensure that the content store version of the document is equivalent to its Whitehall rendered version.

Part of [Trello](https://trello.com/c/oyOq9WtM/369-9-publications-migration-implement-publishing-of-format-to-publishing-api-large-sync-checks-100ish-107-321)

### CODE REVIEW

File: lib/sync_checker/formats/publication_check.rb [hunk 1]
```diff
@@ -57,6 +58,16 @@ def expected_details_hash(edition)
       def rendering_app
         Whitehall::RenderingApp::WHITEHALL_FRONTEND
       end
+
+      def filter_documents_for_locale(edition, locale)
```
Comment (bevanloon) [inline lib/sync_checker/formats/publication_check.rb]: Is it worth extracting this? - basically the same as the method in https://github.com/alphagov/whitehall/pull/2982/files#diff-08abbfe2cb3a245251d671a5077347bb

Comment (gpeng) [inline lib/sync_checker/formats/publication_check.rb]: Yeah... I thought about that. It's a tricky one as one is essentially test code and one is part of the code under test and it seems a bit pointless to test code with itself. We've kind of done a mix of that throughout the sync checks.

Comment (bevanloon) [inline lib/sync_checker/formats/publication_check.rb]: 👍 yep.....that's a good point.

File: app/presenters/publishing_api/publication_presenter.rb [hunk 2]
```diff
@@ -75,7 +75,20 @@ def body
...
+      Whitehall::GovspeakRenderer.new.block_attachments(
+        attachments_for_current_locale,
+        alternative_format_email
+      )
+    end
+
+    def attachments_for_current_locale
+      attachments = item.attachments
+      locales_that_match = Array(I18n.locale.to_s)
+      locales_that_match << "" if I18n.locale == I18n.default_locale
```
Comment (steventux) [inline app/presenters/publishing_api/publication_presenter.rb]: What's the purpose of pushing an empty string here? Does it match attachments with an empty locale on line 90?

Comment (gpeng) [inline app/presenters/publishing_api/publication_presenter.rb]: Yep. The default locale is `:en` but helpfully attachments for the default locale in the whitehall DB have `locale == nil`. It looks like they are all `nil` but selecting on `nil` and `en` seemed like a good idea to be safe.

**Pull request merged**